embulk-input-s3/src/main/java/org/embulk/input/s3/AbstractS3FileInputPlugin.java

```diff
@@ -67,6 +67,11 @@
         @ConfigDefault("null")
         public Optional<String> getAccessKeyId();
 
+        @Config("http_proxy")
+        @ConfigDefault("null")
+        public Optional<HttpProxy> getHttpProxy();
+        public void setHttpProxy(Optional<HttpProxy> httpProxy);
```

Comment:
I'm worried that it can be a kind of compatibility issue. If a user has been setting the HTTP_PROXY env so far, this change may break their behavior.

I thought the best option can be eventually 2. What do you think about taking 3 for the time being, and implementing 2 later?